cmake/TBBBuildInstallScript.cmake

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) 2017-2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This file is a cmake wrapper around existing tools to facilitate
+# building tbb in a way that installs the files
+# consistent with standard directory layouts
+#
+
+#
+# CMAKE_INSTALL_PREFIX  <- Where the installed resources should be placed
+#    ${CMAKE_INSTALL_PREFIX}/lib             <- The library files (both static & shared, and both release & debug)
+#    ${CMAKE_INSTALL_PREFIX}/lib/cmake/TBB   <- Where the cmake package files should be installed
+#    ${CMAKE_INSTALL_PREFIX/include          <- The include files
+# SYSTEM_NAME                                <- The system name for the binaries (i.e. Darwin/Linux/Windows/Android)
+# TBB_BUILD_DIR:PATH                         <- Where temporary object files are to be generated
+# TBB_BUILD_PREFIX:STRING                    <- BUILDPREFIX, or os build prefix
+# TBB_RELATIVE_CMAKE_CONFIG_PATH             <- on installation, relative to CMAEK_INSTALL_PREFIX, where should the cmake package files be placed
+# TBB_COMPILER:STRING                        <- (optional) clang or gcc
+# TBB_BUILD_STATIC:BOOL                      <- (optional) By default static libs are not built.
```

Comment:
Officially TBB doesn't support static libraries. There is a special case for Big Iron platform. So if you'd like to cover this case it would be better to add Big Iron as an option for `SYSTEM_NAME`. In this case `TBB_BUILD_STATIC` and `TBB_BUILD_SHARED` flags are not needed at all.

Reply:
Just for information, "Big Iron" is not really a platform name; it's a common name for mainframe computers and similar HPC systems. Often dynamic (shared) libraries are avoided or prohibited on such systems.

It is true that building and using TBB as a static library is highly discouraged, as it opens the door to have multiple copies of TBB in the same process, with potential negative impact on performance or even correctness of the program.